subprojects/execution/src/main/java/org/gradle/internal/execution/impl/OutputsCleaner.java

```diff
@@ -86,9 +95,12 @@ private void markDirForDeletion(@Nullable File dir) {
         }
     }
 
-    private void cleanupDirectories() throws IOException {
+    public void cleanupDirectories() throws IOException {
         for (File directory = directoriesToDelete.poll(); directory != null; directory = directoriesToDelete.poll()) {
             if (isEmpty(directory)) {
+                if (debugEnabled) {
+                    LOGGER.debug("Deleting stale empty output directory '{}'.", directory.getAbsolutePath());
+                }
```

Comment:
All `File` objects are constructed with absolute paths, you can simply pass `directory` and remove `debugEnabled`.

Reply:
Done.